docs/migration-guides/v2/lib.md

```diff
@@ -0,0 +1,375 @@
+# Migration guide for ably-js v2
+
+Here’s how to migrate from ably-js v1 to v2:
+
+1. [Stop using functionality that v1 deprecated and which v2 removes](#stop-using-v1-deprecated).
+1. [Stop using other functionality that v2 removes](#stop-using-v2-removed).
+1. [Update to v2 and handle its breaking changes](#update-to-v2-and-handle-breaking-changes).
+1. (Optional) [Stop using functionality that v2 deprecates](#stop-using-v2-deprecated).
+1. (Optional) [Take advantage of new features that v2 introduces](#use-v2-new-features).
+
+<h2 id="stop-using-v1-deprecated">Stop using functionality that v1 deprecated and which v2 removes</h2>
+
+Begin by updating to ably-js 1.2.50 or later, to make sure you see the deprecation log messages [described below](#stop-using-v1-deprecated-all-users).
+
+Now, you need to stop using the functionality that is deprecated by v1 and which is removed in v2. Here we explain how.
+
+The changes below are split into those that affect all users, and those that only affect TypeScript users.
+
+<h3 id="stop-using-v1-deprecated-all-users">Changes that affect all users</h3>
```

Comment:
not gonna lie, on my first read of this my heart skipped a beat when i saw "changes that affect all users" followed by a long wall of text 😬 not sure exactly what to suggest but i think we could do better to make it immediately clear that these are breaking changes to individual features which users may or not be using

Reply:
Yeah, that's a good point. I've:

- changed the intro to each section to say:
 
  > The changes below are split into:
  > 
  > - general changes
  > - changes that only affect TypeScript users
- changed "All users" headings to "General changes"
- added the sentence "If you’re not using any of the deprecated functionality described below, then this section does not affect you." to the deprecated v1 functionality intro
- added the sentence "Some of these changes are only relevant if you’re using specific features of the library. The guidance below makes it clear when this is the case." to the v2 breaking changes section.

Reply:
Resolving since Owen is away